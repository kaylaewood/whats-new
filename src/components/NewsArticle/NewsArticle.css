.newsArticle {
  border: 1px solid white;
}

.newsImage {
  width: 100px;
}
