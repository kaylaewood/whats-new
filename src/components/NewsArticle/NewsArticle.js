import React from 'react';
import './NewsArticle.css';

function NewsArticle(props) {
  return (
    <div className='newsArticle'>
      <img className='newsImage' src={props.image} alt={`${props.headline}'`} />
      <div className='newsInfo'>
        <h2>{props.headline}</h2>
        <p>{props.description}</p>
      </div>
      <div className='link-box'>
        <a href={props.link}>CLICK TO GO TO ARTICLE</a>
      </div>
    </div>
  )
}
export default NewsArticle;
