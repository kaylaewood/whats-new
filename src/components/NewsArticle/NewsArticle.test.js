import React from 'react';
import ReactDOM from 'react-dom';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow(<NewsArticle
      image = {'image-url'}
      headline = {'Headline Text'}
      description = {'This is the description.'}
      link = {'link-url'}
    />);
    expect(wrapper).toMatchSnapshot();
  })
});
