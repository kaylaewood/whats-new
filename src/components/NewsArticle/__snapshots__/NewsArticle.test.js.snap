// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should correctly match the snapshot with all the data passed correctly 1`] = `
<div
  className="newsArticle"
>
  <img
    alt="Headline Text'"
    className="newsImage"
    src="image-url"
  />
  <div
    className="newsInfo"
  >
    <h2>
      Headline Text
    </h2>
    <p
      className="article-description"
    >
      This is the description.
    </p>
  </div>
  <div
    className="link-box"
  >
    <a
      href="link-url"
    >
      CLICK TO GO TO ARTICLE
    </a>
  </div>
</div>
`;
